### DISCUSSION TITLE
HELP! Linux Surface updated and now the machine is stuck at boot

### DISCUSSION THREAD

**Discussion by longtry** [linux-surface/linux-surface#1194] [Q&A]
I installed Linux Surface over Mint Vera a month ago. Just today, Mint's auto update gave me a notification about the newest LS update, and I installed it (FYI, it also told me to install Mint Victoria, but I didn't do). At some point during the installation, a dialog box appeared that prompted me to set a password for some SecureBoot's MOK patch during the next boot. I typed a password, and it proceeded normally. Then, tonight when I reboot, my Surface Book 2 just stuck at the very 1st screen, the 4-square Window logo. After a minute of waiting, nothing happened. I rebooted many times and tried a few combinations of key pressing to attempt to open UEFI settings, to no avail. It just stuck dead. Any ideas? Thanks!
Oh, btw, my previous UEFI boot sequence is: USB - Linux - Windows. Knowing this, I tried inserting what I remember as a Linux USB into it, but no change at all.

**Comment by qzed** [comment]
You should generally always be able to enter UEFI by pressing and holding volume up (the one next to the power button) while powering on the device. If that doesn't work, try holding both power button and volume up while the device is on until it reboots into the UEFI menu.

The issue of being stuck could also be related to #1162

**Reply by longtry** [reply]
Ah yes, it worked, tks! I accessed UEFI and the correct boot seq is Linux - USB - Win. Leaving it, the machine will get stuck. I disable the Linux option, and _now_ the MOK insert screen appears at boot. I installed it, but then the boot process says system error. I re-enable the Linux option and now it boots OK, with the updated LS intact.

So it looks like (during Mint-based auto update) somehow your MOK driver programming is attached to Windows... right? Although the problem is solved, that's the easiest way to make sense of the situation.

**Comment by Ramen-LadyHKG** [comment]
> > Post some screenshots here.
> > ```
> > 1. Bitlocker setting page in Windows
> > 
> > 2. run `sudo blkid` in Linux terminal
> > 
> > 3. the warning sign you're referring to
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > **Issue A)** I'm not sure that you've mount Windows partition before. In order to mount an encrypted system drive (specifically C drive), you need to use recovery key.
> > **Issue B)** The `394.651.768.0`firmware, AFAIK, it only affects booting.
> > Since you're using mint, you're most likely unable to boot before downgrade back to `392`, unless you're NOT using Grub or the Grub isn't Redhat/Canonical issued.
> 
> OK, here goes.
> 
>     1. The Bitlocker page is now all lovey-dovey, because I enabled it some days ago. Before that, it was off.
> 
>     2. [blkid](https://imgur.com/a/OqdfO4o) here.

@longtry Your partition table is a bit confusing to me
I don't understand why there are 2 bitlocker partitions and 1 ntfs partition.

My guess is your Windows partition is not encrypted but you have 2 encrypted partition to store data?

Please explain what `partitions 345` are? or show `Disk Management` in Windows

> 
>     3. [Exclamation-in-triangle](https://imgur.com/a/QizWZ2a) here (not a true scrshot tho, because as said in 1., I enabled it and they're no longer in Win Explorer.
> 
> 
> A. Yes, my Mint mounted C & D:\ all the time, and I read & saved files on them w/o probs. It was after the auto-update of Linux Surface on Mint that I encountered the dead-Win screen at bootup, and after the measures taken to get pass that state (described in my #1194 post) that Mint began to fail to mount the drives.
> 
> B. idk what grub is for sure, I just installed Mint 2 months ago using mostly the default setup guide. Yet I know for sure that I'm having [394 running](https://imgur.com/a/lDzm6DN) right now, and I can boot into LS Mint also quite easily - after that scare a week ago, that is.

I'm not sure what the Windows C: drive Screenshot means

On Linux Mint,
Can you try mount the partition in the terminal with verbose?
for example,
`sudo mount -v /dev/nvme0n1p4`
`sudo cryptsetup bitlkOpen /dev/nvme0n1p3 WinM`
`sudo mount -v /dev/mapper/WinM /mnt`

**Comment by longtry** [comment]
> My guess is your Windows partition is not encrypted but you have 2 encrypted partition to store data? Please explain what partitions 345 are?

Your guess is as good as mine. However, I got the win's [disk man](https://imgur.com/a/LE6wSzC). The partition 6 is Mint.

> Can you try mount the partition in the terminal with verbose?

Sure, [here it is](https://imgur.com/jaSzyzi). I tried putting in the passphrase for D:\, so it's likely that 0n1p3 is C:\

**Reply by Ramen-LadyHKG** [reply]
Your first screenshot lack of the graphic below which is way more helpful

The bitlocker password for C:\ is likely masked by Microsoft if it was encrypted automatically by Windows.
Because Windows uses TPM to unlock C: instead of the user password.

In that case, you need to use recovery password to unlock the C:\ in Linux.
> you can find the recovery password in [Microsoft Account](https://aka.ms/myrecoverykey) if you've logged in on Windows

> other wise, you need to boot into Windows and go to the bitlocker manager, and click the `output recovery key` button

For your other drive, I'm not sure how you encrypted the D:\ with password or System like the C:\ case above.
> You may also find another  recovery key with the method above.

once you gather all the recovery key, you can use it as an password to unlock the partition
> recovery key looks like this `xxxxxx-xxxxxx-xxxxxx-xxxxxx-xxxxxx-xxxxxx-xxxxxx`

use the command I provided above try to mount it.
if you successfully mount it, you can setup `crypttab` `fstab` for automatically unlock when boot.

ask me for further instructions
